src/example-types/tagsQuery.js

```diff
@@ -0,0 +1,89 @@
+let _ = require('lodash/fp')
+let { parens } = require('futil-js')
+let Combinatorics = require('js-combinatorics')
+
+let wordPermutations = word =>
+  _.map(_.join(' '), Combinatorics.permutation(word.split(' ')).toArray())
+
+let limitResultsToCertainTags = tags => !!_.find('onlyShowTheseResults', tags)
+
+let wrapIf = _.curry(
+  (pre, post, text, shouldWrap) => (shouldWrap ? `${pre}${text}${post}` : text)
+)
+let quoteIf = wrapIf('"', '"')
+
+let quoteAndTilde = _.curry(
+  (tag, text) =>
+    quoteIf(text, tag.isPhrase) +
+    (tag.misspellings || tag.distance ? '~' : '') +
+    (tag.distance || '')
+)
+
+let escapeSpecialChars = text =>
+  text.toString().replace(/([!*+\-=<>&|()[\]{}^~?:\\/"])/g, '\\$1')
```

Comment:
Is this from the lucene docs? We should include a link to where this list came from.

Reply:
Also, it doesn't seem to handle the current case of explicitly escaped quotes by end users

Reply:
https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-query-string-query.html#_reserved_characters